src/collection.jl

```diff
@@ -115,30 +194,252 @@ end
 
 
 """
-    filenames(df::DataFrame)
+    filenames(collection)
 
 Generator for filenames of entries in data frame.
+
+Iterates over `collection` using each `path`.
+
+# Examples
+```julia
+collection = fitscollection("~/data/tekdata")
+for path in filenames(collection)
+    @assert path isa String
+    println(path)
+end
+
+# output
+"~/data/tekdata/tek001.fits"
+"~/data/tekdata/tek002.fits"
+...
+```
 """
 function filenames end
 
 # generator for filenames specified by data frame (i.e. path of file, hdu etc.)
-@resumable function filenames(df::DataFrame)
-    for row in eachrow(df)
+@resumable function filenames(collection)
+    for row in eachrow(collection)
         @yield row.path
     end
 end
 
 
 """
-    images(df::DataFrame)
+    ccds(collection)
+
+Generator for `CCDData`s of entries in data frame.
+
+Iterates over `collection` using each `path` and `hdu` to load data using [`FITSIO.FITS`](http://juliaastro.github.io/FITSIO.jl/latest/api.html#FITSIO.FITS) into a [`CCDData`](@ref).
```

Comment:
```suggestion
Iterates over `collection` using each `path` and `hdu` to load data into a [`CCDData`](@ref).
```
I don't seen the need to include that extra detail at the sacrifice of grammatical flow